### DISCUSSION TITLE
Details on cache eviction

### DISCUSSION THREAD

**Discussion by Nohac** [Azure/azure-storage-fuse#998] [Q&A] [answered]
Hi. I have a few questions regarding how the cache eviction works.
1. Is it possible to set an infinite cache timeout, and only evict based on disk usage?
2. How does the algorithm decide what to evict, is it possible to set it up to only evict files that's not been accessed for a long time?

A use case for this is caching a RO data set for machine learning, the data never really change, but some data will eventually stop being used (accessed) due to either being bad or outdated data.

**Comment by bhansalivikas** [comment, marked as answer]
1. Is it possible to set an infinite cache timeout, and only evict based on disk usage?
 - For this you might need to put up a large value for file-cache timeout something like a month or so.
2. How does the algorithm decide what to evict, is it possible to set it up to only evict files that's not been accessed for a long time?
- Blobfuse2 cache eviction logic is already LRU based (based on what file was last closed by user application). However, if the configured cache size or disk is getting full then it may not wait for the eviction timeout and start early.
